### DISCUSSION TITLE
Getting horizontal slices for whitespace analysis

### DISCUSSION THREAD

**Discussion by mperitz** [ImageMagick/ImageMagick#4978] [Help]
Hello!  I am using version 6.9.11-60 through the CLI.

I want to analyze a set of png screenshots.  Each screenshot is created by a program, and you can set the height and width of the screenshot beforehand.  However, most people just leave it on the default full page size, which results in files that are really big for no reason (see attachment for a contrived example).  So, I want to be able to see how many images are unnecessarily large by seeing how much whitespace is in each image.

I wrote a script that calculates the total percentage of pixels that are white, but since the background color of each screenshot is white, even images that have been appropriately sized come out to at least 90% whitespace based on pixel count.

What would give me a better signal is if I could split up the image into <WIDTH>x1 slices, where <WIDTH> represents the width of the image.  For example, say my image is 1024x768 pixels, each slice would be 1024x1 pixels in area (for performance reasons I could probably expand this to 10 or even 100 pixels).  Analyzing each slice for its percent whitespace and figuring out the percent of slices that are more than, say, 97% whitespace would be a really good signal that the image in question is not appropriately sized.

I'm new to this and am not really sure how to go about doing this.  Can anyone help me out?  How can I accomplish this, if at all?  Is this even the right way to go about this?

Thank you for any and all help!

![Screen Shot 2022-03-22 at 11 24 12 AM](https://user-images.githubusercontent.com/24237476/159518149-1c69e6db-c0bc-4591-8c2b-21b339c55161.png)
!

**Comment by fmw42** [comment]
You can use -crop to separate an image into equal size regions.  

convert image.suffix -crop 1024x1 +repage output.suffix

will make the slices that you want.  See tile cropping at https://legacy.imagemagick.org/Usage/crop/#crop_tile

**Comment by fmw42** [comment]
You could average this down to one column. Then look for the largest totally white contiguous section in the one column. Find start and stop sections of white to get the regions of white.

`convert image.suffix -scale "1x!" one_column.suffix`

or

`convert image.suffix -scale "1x!" txt:`

will list out each pixel coordinate and its color values

**Reply by mperitz** [reply]
Thanks!  This works great for my needs.

**Reply by mperitz** [reply]
What if I wanted a "row" slice instead of a column?  I tried reversing the 1x! to !x1 but that did not work.

**Comment by fmw42** [comment]
Use "x1!"
